### DISCUSSION TITLE
Difference between `require(...).setup()` in `config` vs. in `init.lua`

### DISCUSSION THREAD

**Discussion by dongcarl** [wbthomason/packer.nvim#1073] [Q&A] [answered]
I noticed for some packages like `Comment.nvim`, they require that a `require(...).setup()` be made both in the packer `config` section:

```lua
use {
    'numToStr/Comment.nvim',
    config = function()
        require('Comment').setup()
    end
}
```

and in `init.lua`:

```lua
require('Comment').setup()
```

I experimented with removing the `init.lua` line, and it stopped working.

So I guess my question is, what is the difference between specifying a config hook in a `packer` `use` and specifying the same thing in `init.lua`?

**Comment by nyngwang** [comment]
Did you import your first code segment from init.lua? If so, I think the setup inside `init.lua` can be removed. I never put any `setup(...)` inside `init.lua` and have a working neovim still.

**Comment by gegoune** [comment, marked as answer]
There isn't really a difference. If you put it in init.lua file it will be executed whenever that file runs. If you put it in packer's config it will run through packer. End result is going to be the same - jut pick one. Some prefer to keep configuration together with plugin declaration (packer's config), some specify list of plugins alone and configure/setup those plugin elsewhere in their configs. Like in after/plugin/* files or init.lua. It is a matter of preference really.
